src/util/version.ts

```diff
@@ -58,11 +58,23 @@ export function variants(version: string) {
   const kabobBuild = version.replace(/[+]/g, "-");
   // todo: add any other platform specific variants here.
   return {
+    version_default: version,
     version_dotnet: kabobBuild,
     version_docker: kabobBuild,
   };
 }
 
+export function variantByKey(version: string, key: string) {
+  const { version_dotnet, version_docker, version_default } = variants(version);
+  if(key === "version_dotnet") {
+    return version_dotnet;
+  }
+  if(key === "version_docker") {
+    return version_docker;
+  }
+  return version_default;
+}
```

Comment:
since your key is just the object property you could probably reduce this to:

```ts
export function variantByKey(version: string, key?: string) {
  return variants(version)[key ?? "variant_default"]
}
```

Also what do you think about creating an enum?

```ts
export enum VariantKey {
  Default = "variant_default",
  Dotnet = "variant_dotnet",
  Docker = "variant_docker"
}
```